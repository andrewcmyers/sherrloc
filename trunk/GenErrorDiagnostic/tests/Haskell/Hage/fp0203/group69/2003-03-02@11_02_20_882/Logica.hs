module Logica where
data Prop
  = En [Prop]
  | Of [Prop]
  | Niet Prop
  | Prop :-> Prop
  | Var String
  | Bool Bool

type Bedeling = [String]
elemBy        :: (a -> a -> Bool) -> a -> [a] -> Bool
elemBy = undefined
eqString      :: String -> String -> Bool 
eqString = undefined

evalueer :: Prop -> Bedeling -> Bool
evalueer (En propositie) bedeling   = and (map (\prop -> evalueer prop bedeling) propositie)
evalueer (Of propositie) bedeling   = or (map (\prop -> evalueer prop bedeling) propositie)
evalueer (Var string) propositie    = elemBy eqString string propositie
evalueer (Bool bool) _              = bool
evalueer (Niet propositie) bedeling = not (evalueer propositie bedeling)
evalueer (prop1 :-> prop2) bedeling = or (not (evalueer prop1 bedeling)):(evalueer prop2 bedeling)

-- the parameter of 'or' should be: [not (evalueer prop1 bedeling),evalueer prop2 bedeling]
-- 22,39-98
