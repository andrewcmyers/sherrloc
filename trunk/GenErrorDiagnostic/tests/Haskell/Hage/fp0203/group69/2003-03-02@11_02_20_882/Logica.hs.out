Compiling ./group69/2003-03-02@11:02:20:882/Logica.hs
(17,42): Type error in application
 expression       : or (not (evalueer prop1 bedeling))
 function         : or
   type           : [Bool] -> Bool
 1st argument     : (not (evalueer prop1 bedeling))
   type           : Bool  
   does not match : [Bool]

(17,73): Type error in infix application
 expression       : or (not (... prop1 ...)) : (evalueer prop2 bedeling)
 operator         : :
   type           : a -> [a]  -> [a] 
   does not match : b -> Bool -> Bool

Compilation failed with 2 errors
